src/server/game/Server/WorldSocket.cpp

```diff
@@ -358,6 +358,8 @@ WorldSocket::ReadDataHandlerResult WorldSocket::ReadDataHandler()
         }
         case CMSG_KEEP_ALIVE:
             LogOpcodeText(opcode, sessionGuard);
+            if (_worldSession)
```

Comment:
This is very bad - you are accessing _worldSession while sessionGuard is not locked